### PR TITLE
Add ACR Install tests

### LINKED ISSUE #1559: Add more tests for ACR
### Summary of the new feature / enhancement

- Install tests for ACR
- Publish tests for ACR

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Adding test suite for `Install-PSResource` from ACR server. 

Resolves #1559 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+    <# TODO:  enable after implementing script functionality
+    It "Install specific script resource by name" {
+        Install-PSResource -Name $testScriptName -Repository $ACRRepoName -TrustRepository
+        $pkg = Get-InstalledPSResource $testScriptName
+        $pkg.Name | Should -Be $testScriptName
+        $pkg.Version | Should -Be "1.0.0"
+    }
+    #>
```
Comment (adityapatwardhan) [inline test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1:56]: We might need a property in the manifest for package type module / script

File: test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+    # TODO: Update this test and others like it that use try/catch blocks instead of Should -Throw
+    It "Should not install resource with incorrectly formatted version such as exclusive version (1.0.0.0)" {
+        $Version = "(1.0.0.0)"
+        try {
+            Install-PSResource -Name $testModuleName -Version $Version -Repository $ACRRepoName -TrustRepository -ErrorAction SilentlyContinue
+        }
+        catch
+        {}
+        $Error[0].FullyQualifiedErrorId | Should -be "IncorrectVersionFormat,Microsoft.PowerShell.PSResourceGet.Cmdlets.InstallPSResource"
```
Comment (adityapatwardhan) [inline test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1]:
```suggestion
        { Install-PSResource -Name $testModuleName -Version $Version -Repository $ACRRepoName -TrustRepository -ErrorAction SilentlyContinue } | Should -Throw -ErrorId "IncorrectVersionFormat,Microsoft.PowerShell.PSResourceGet.Cmdlets.InstallPSResource"
```

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1 (hunk 1 above)
Comment (alerickson) [reply test/InstallPSResourceTests/InstallPSResourceACRServer.Tests.ps1:56]: Opened up an issue here to track this: https://github.com/PowerShell/PSResourceGet/issues/1568.  I'll add this comment there as well

**Review by adityapatwardhan — APPROVED** [review]

**Review by anamnavi — APPROVED** [review]
LGTM

**Pull request merged**